data/data-pipeline/data_pipeline/comparison_tool/src/utils.py

```diff
@@ -0,0 +1,356 @@
+import pathlib
+import pandas as pd
+import xlsxwriter
+
+from data_pipeline.score import field_names
+from data_pipeline.etl.sources.census.etl_utils import get_state_information
+
+# Some excel parameters
+DEFAULT_COLUMN_WIDTH = 18
+# the 31 is a limit from excel on how long the tab name can be
+MSFT_TAB_NAME_LIMIT = 31
+
+# FIPS information
+DATA_PATH = pathlib.Path(__file__).parents[2] / "data"
+FIPS_MAP = (
+    get_state_information(data_path=DATA_PATH)
+    .set_index("fips")["state_abbreviation"]
+    .to_dict()
+)
+
+
+def read_file(
+    file_path: str, columns: list, geoid: str = field_names.GEOID_TRACT_FIELD
+) -> pd.DataFrame:
+    """Reads standardized csvs in
+
+    Parameters:
+        file_path: the file to read
+        columns: the columns to include
+        geoid: the geoid column name (if we change this in usa.csv, will need to change this slightly)
+
+    Returns:
+        dataframe that has been read in with geographic index
+    """
+    assert (
+        geoid == field_names.GEOID_TRACT_FIELD
+    ), f"Field name specified for geoid is incorrect. Use {field_names.GEOID_TRACT_FIELD}"
+    return pd.read_csv(
+        file_path, usecols=columns + [geoid], dtype={geoid: str}
+    ).set_index(geoid)
+
+
+def produce_summary_stats(
+    joined_df: pd.DataFrame,
+    comparator_column: str,
+    score_column: str,
+    population_column: str,
+    geoid_column: str = field_names.GEOID_TRACT_FIELD,
+) -> pd.DataFrame:
+    """Produces high-level overview dataframe
+
+    Parameters:
+        joined_df: the big df
+        comparator_column: the column name for the comparator identification bool
+        score_column: the column name for the CEJST score bool
+        population_column: the column that includes population count per tract
+        geoid_column: the geoid10_tract column
+
+    Returns:
+        population: the high-level overview df
+    """
+    # I actually think this makes sense to be out of all tracts
```

Comment:
nit: could you make this comment a bit more formal and self-explanatory? I'm not sure I understand what you mean here.

Reply:
Yup!